packages/taiko-client/driver/preconf_blocks/api.go

```diff
@@ -69,6 +71,29 @@ func (s *PreconfBlockAPIServer) BuildPreconfBlock(c echo.Context) error {
 		return s.returnError(c, http.StatusBadRequest, errors.New("executable data is required"))
 	}
 
+	parent, err := s.rpc.L2.HeaderByHash(c.Request().Context(), reqBody.ExecutableData.ParentHash)
+	if err != nil {
+		return s.returnError(c, http.StatusInternalServerError, err)
+	}
+
+	if parent.Number.Uint64() < s.latestBlockIDSeenInEvent {
+		log.Warn("The parent block ID is smaller than the latest block ID seen in event",
+			"parentBlockID", parent.Number.Uint64(),
+			"latestBlockIDSeenInEvent", s.latestBlockIDSeenInEvent,
+		)
+		return s.returnError(c, http.StatusBadRequest,
```

Comment:
seems https://github.com/taikoxyz/taiko-mono/blob/main/packages/taiko-client/driver/chain_syncer/event/blocks_inserter/pacaya.go#L293 will also handle this issue?